buildtools/DevConfigGenerator/DevConfigGenerator/Program.cs

```diff
@@ -0,0 +1,209 @@
+
+    AnsiConsole.Markup($"DevConfig file written to: [green]{devConfigFilePath}[/]\n");
+    AnsiConsole.WriteLine(File.ReadAllText(devConfigFilePath));
+}
+
+
+void CoreChange()
+{
+    var change = AnsiConsole.Prompt(new TextPrompt<string>("Enter the change message for the [green]core[/] change:"));
+
+    var newCoreConfig = false;
+    if (devConfig.Core == null)
+    {
+        newCoreConfig = true;
+        devConfig.Core = new CoreChange();
+    }
+
+    devConfig.Core.Messages.Add(change);
+
+    devConfig.Core.Type = AskChangeType(devConfig.Core.Type);
+
+    var textPrompt = new TextPrompt<bool>("Should service packages be updated to use new Core?")
+            .AddChoice(true)
+            .AddChoice(false)
+            .WithConverter(choice => choice ? "y" : "n");
+
+    // If added a second change message to core then default the update service minimum to the value choosen for the previous entry.
+    if (!newCoreConfig)
+    {
+        textPrompt.DefaultValue(devConfig.Core.UpdateServiceMinimum);
+    }
+
+    var updateServiceMinimum = AnsiConsole.Prompt(textPrompt);
+
+    devConfig.Core.UpdateServiceMinimum = updateServiceMinimum;
+}
+
+void GeneratorChange()
+{
+    AnsiConsole.Markup("Generator changes are saved as [green]Core changes[/] with update service minimum set to [green]true[/].\n");
+    var change = AnsiConsole.Prompt(new TextPrompt<string>("Enter the change message for the [green]generator[/] change:"));
+
+    if (devConfig.Core == null)
+    {
+        devConfig.Core = new CoreChange();
+    }
+
+    devConfig.Core.Messages.Add(change);
+    devConfig.Core.UpdateServiceMinimum = true;
+
+    devConfig.Core.Type = AskChangeType(devConfig.Core.Type);
+}
+
+void ServiceChange()
+{
+    var serviceNames = new List<string>();
+    foreach(var serviceDirectory in Directory.GetDirectories(Path.Combine(FindRepoRoot(), "sdk", "src", "Services")))
+    {
+        serviceNames.Add(new DirectoryInfo(serviceDirectory).Name);
+    }
+
+    var searchQuery = AnsiConsole.Ask<string>("What service was changed (The name should match directory name in the sdk/src/Services parent directory)?");
+
+
+    string serviceName;
+    if (serviceNames.FirstOrDefault(x => string.Equals(x, searchQuery, StringComparison.OrdinalIgnoreCase)) != null)
+    {
+        serviceName = serviceNames.FirstOrDefault(x => string.Equals(x, searchQuery, StringComparison.OrdinalIgnoreCase))!;
+    }
+    else
+    {
+        var topMatches = FuzzySharp.Process.ExtractTop(searchQuery, serviceNames)
+                                .Select(result => result.Value)
+                                .ToList();
+
+        serviceName = AnsiConsole.Prompt(
+            new SelectionPrompt<string>()
+                .PageSize(10)
+                .AddChoices(topMatches));
+    }
+
+
+    var service = devConfig.Services.FirstOrDefault(x => x.Name == serviceName);
+    if (service == null)
+    {
+        service = new Service { Name = serviceName };
```

Comment:
When would this case be valid? It seems like it would either be a mistake name entered that has no match or trying to specify a new service which would have a folder name. Looks like this could lead to invalid devconfigs.

Reply:
This is a check to see if this is the first time the service has been added to the dev config file. If I add a second change log entry for a service I already add I want the same service add to the list of change services.  The first time a change log entry is added for a service this will be null and added to the list of changed services.

This isn't adding a new service to our list of supported service or anything. I added a comment to make it more clear.